src/tests/GfxBatchHbaoTest.cpp

```diff
@@ -0,0 +1,413 @@
+#include <Magnum/ImageView.h>
+#include <Magnum/Math/Angle.h>
+#include <Magnum/Math/Color.h>
+#include <Magnum/Math/Matrix4.h>
+#include <Magnum/MeshTools/Compile.h>
+#include <Magnum/PixelFormat.h>
+#include <Magnum/Shaders/PhongGL.h>
+#include <Magnum/Trade/AbstractImporter.h>
+#include <Magnum/Trade/ImageData.h>
+#include <Magnum/Trade/MeshData.h>
+
+#include "esp/gfx_batch/Hbao.h"
+
+#include "configure.h"
+
+namespace {
+
+namespace Cr = Corrade;
+namespace Mn = Magnum;
+using namespace Mn::Math::Literals;
+
+constexpr const char* baseTestFilename = "van-gogh-room";
+
+const std::string SourceTestPlyDir =
+    Cr::Utility::Path::join(SCENE_DATASETS, "habitat-test-scenes");
+
+const std::string TestHBAOImageDir =
+    Cr::Utility::Path::join(TEST_ASSETS, "hbao_tests");
+
+constexpr Mn::Vector2i Size{320, 240};
+
+constexpr const float aspectRatio = float(Size.x()) / Size.y();
+
+struct TestDataType;
+struct Projection;
+
+struct GfxBatchHbaoTest : Mn::GL::OpenGLTester {
+  explicit GfxBatchHbaoTest();
+  void generateTestData();
+  void testPerspective();
+
+  void testPerspectiveFlipped();
+  void testOrthographic();
+
+  void testHBAOData(const TestDataType& data,
+                    const std::string& filename,
+                    const Projection& proj,
+                    bool flipped);
+};
+
+struct Projection {
+  Mn::Matrix4 projection;
+  const std::string sourceColorFilename;
+  const std::string sourceDepthFilename;
+};
+
+const Projection perspectiveData{
+    Mn::Matrix4::perspectiveProjection(45.0_degf, aspectRatio, 0.1f, 100.0f),
+    Cr::Utility::formatString("{}.color.png", baseTestFilename),
+    Cr::Utility::formatString("{}.depth.exr", baseTestFilename)};
+
+const Projection flippedPerspectiveData{
+    Mn::Matrix4::perspectiveProjection(
+        2.0f * Mn::Deg((Mn::Math::atan(Mn::Math::tan(Mn::Rad(45.0_degf) * 0.5) /
+                                       aspectRatio))),
+        1.0f / aspectRatio,
+        0.1f,
+        100.0f),
+    Cr::Utility::formatString("{}.color.png", baseTestFilename),
+    Cr::Utility::formatString("{}.depth.exr", baseTestFilename)};
+
+const Projection orthographicData{
+    Mn::Matrix4::orthographicProjection(
+        Magnum::Vector2(4.0f * aspectRatio, 4.0f),
+        0.1f,
+        100.0f),
+    Cr::Utility::formatString("{}.color-ortho.png", baseTestFilename),
+    Cr::Utility::formatString("{}.depth-ortho.exr", baseTestFilename)};
+
+const struct TestDataType {
+  const char* name;
+  const char* filename;
+  bool classic;
+  esp::gfx_batch::HbaoFlags flags;
+  Mn::Float intensity, radius, blurSharpness;
+  Mn::Float maxThreshold, meanThreshold;
+} TestData[]{
+    // Perspective projection
+    {"classic, defaults", "hbao-classic", true,
+     esp::gfx_batch::HbaoConfiguration{}.flags(),
+     esp::gfx_batch::HbaoConfiguration{}.intensity(),
+     esp::gfx_batch::HbaoConfiguration{}.radius(),
+     esp::gfx_batch::HbaoConfiguration{}.blurSharpness(), 0.0f, 0.0f},
```

Comment:
Heh, these intensity / radius / shaprness copypastes are kinda annoying. What about just storing a "multiplier" here, have it `1.0f` most of the time, and then multiply in the test directly?

Reply:
Yeah, rearranged that stuff now to be a bit less wordy.